debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/ObjectFieldInfo.java

```diff
@@ -398,21 +400,36 @@ boolean isInstanceFieldBackfillEligible() {
 						if (!valueTypeHelper.isJ9FieldIsFlattened(fieldClass, f)) {
 							instanceObjectCount += 1;
 							totalObjectCount += 1;
-						} else if (valueTypeHelper.isJ9ClassLargestAlignmentConstraintDouble(fieldClass)) {
-							UDATA doubleSize = fieldClass.totalInstanceSize();
-							if (valueTypeHelper.classRequires4BytePrePadding(fieldClass)) {
-								doubleSize = doubleSize.sub(U32.SIZEOF);
-							}
-							size = Scalar.roundToSizeofU64(doubleSize).intValue();
-							totalFlatFieldDoubleBytes += size;
-						} else if (valueTypeHelper.isJ9ClassLargestAlignmentConstraintReference(fieldClass)) {
-							size = Scalar.roundToSizeToFJ9object(fieldClass.totalInstanceSize()).intValue();
-							totalFlatFieldRefBytes += size;
-							setPotentialFlatObjectInstanceBackfill(size);
 						} else {
-							size = fieldClass.totalInstanceSize().intValue();
-							totalFlatFieldSingleBytes += size;
-							setPotentialFlatSingleInstanceBackfill(size);
+							boolean forceDoudbleAlignment = false;
+							if (fj9object_t_SizeOf == U32.SIZEOF) {
+								UDATA instanceSize = fieldClass.totalInstanceSize();
+								UDATA doubleSize = new UDATA(U64.SIZEOF);
+								if (valueTypeHelper.classRequires4BytePrePadding(fieldClass)) {
+									instanceSize = instanceSize.sub(U32.SIZEOF);
+								}
+								forceDoudbleAlignment = modifiers.allBitsIn(J9JavaAccessFlags.J9AccVolatile) && (instanceSize.eq(doubleSize));
+							} else {
+								forceDoudbleAlignment = true;
+							}
+							if (valueTypeHelper.isJ9ClassLargestAlignmentConstraintDouble(fieldClass)
+								|| forceDoudbleAlignment
+							) {
```

Comment:
Same comments here as above.

Reply:
Fixed.

Reply:
There's still the issue of using `J9ObjectHelper.compressObjectReferences`.